services/authorization/src/main/kotlin/AuthorizationService.kt

```diff
@@ -105,13 +105,22 @@ interface AuthorizationService {
     suspend fun ensureSuperuserAndSynchronizeRolesAndPermissions()
 
     /**
-     * Add the [role] for the hierarchy element with the given [hierarchyId] to the user with the given [username].
+     * Add the [role] for the hierarchy element with the given [hierarchyId] to the user with the given [username]. If
+     * the user already has another role for the same hierarchy element, it will be replaced by the new [role].
      */
-    suspend fun <ID : HierarchyId> addUserRole(username: String, hierarchyId: ID, role: Role<ID>)
+    suspend fun <TYPE : Role<TYPE, ID>, ID : HierarchyId> addUserRole(
```

Comment:
If there is now only a single role, the names `addUserRole` and `removeUserRole` are maybe a bit misleading. Should these methods be renamed to something like `setUserRole` and `unsetUserRole`?

Reply:
Good point, if that's ok I will add this to my todo list for the follow-up PR to avoid the rebase work. I think `remove` is still ok but `add` needs to be replaced.